cmd/agent/app/builder.go

```diff
@@ -190,13 +231,29 @@ func (b *Builder) GetProcessors(rep reporter.Reporter, mFactory metrics.Factory)
 	return retMe, nil
 }
 
+func (b *Builder) useTChannelReporter() bool {
+	// if we don't have credentials, we use the tchannel reporter
+	// if we have an auth token or a pair of username+password, we should use the http reporter
+	return b.AuthToken == "" && (b.Username == "" || b.Password == "")
```

Comment:
I prefer not to introduce these indirect conditions. We should have a cmd line argument telling which reporter to use (default to tchannel). Then HTTP reporter can in addition check the other params.

Reply:
A counter-argument is that we keep the actual implementation hidden behind the requirements: if we see auth data, we know we need to perform auth and we shall use whatever transport provides this feature. Once we change to gRPC, this distinction should disappear.

Reply:
yes, that's precisely the hidden magic that goes against Go's "no magic" principle - adding an argument changes which transport is used, which port needs to be configured, etc. I would much rather be explicit. When we implement gRPC we can revisit.